cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -179,14 +179,43 @@ func getKubeMasterUrl() (string, error) {
 	return parsedUrl.String(), nil
 }
 
+func getDefaultMasterUrl(use_ro_port bool) (string, error) {
```

Comment:
I find this logic unnecessarily complex. I prefer the original flag with the ability to pass env variables.